engine/dlib/src/dlib/log.cpp

```diff
@@ -665,7 +665,7 @@ void LogInternal(LogSeverity severity, const char* domain, const char* format, .
         n += vsnprintf(str_buf + n, dmLog::MAX_STRING_SIZE - n, format, lst);
     }
 
-    if (n < dmLog::MAX_STRING_SIZE)
+    if (n + 1 < dmLog::MAX_STRING_SIZE)
```

Comment:
No comment here? What was the issue?
Did you catch this in the tests?

Reply:
The +1 is to guarantee that there's space in the string for the "\n" appended by `dmSnPrintf` below. If there isn't space and `dmSnPrintf` is called, it would return -1 to signal truncation, which would then be added to `n`, causing the last char to be overwritten if another string were to be printed to `str_buf + n`.

No test — I spotted this as I was looking over all uses of `dmSnPrintf` (since I was fixing one misuse of `dmSnPrintf` return value, I thought I might as well fix all instances of the same misuse).

For this, would you expect a comment in the PR or a comment in the code?

Reply:
Well, yes, since it's old untouched code, it's curious that it's suddenly in need of a +1.
And, it also tells us that we didn't have a test for it.

Reply:
I've added two tests to verify:

1. Logging a message with the maximum allowed length doesn't truncate
2. Logging a message with the maximum allowed length + 1 does truncate

To make the second test pass, I had to change this code again. No more `n + 1 < dmLog::MAX_STRING_SIZE`, since if `n` is `MAX_STRING_SIZE - 1` we still need to increment `n`, even though no character was appended, in order for the truncation to trigger with the next if-statement.

I added a code comment to clarify reasoning.